File: #169_Majority_Element/Solution.java
public class Solution {
    public int majorityElement(int[] nums) {
    	/* Majority Vote Algorithm */
        /* assume that majority element does exist */
        int result = nums[0], count = 1;
        for (int i = 1; i < nums.length; i++) {
        	if (nums[i] == result) count++;
        	else if (count == 0) {
        		result = nums[i];
        		count++;
        	}
        	else count--;
        }
        return result;
    }
}

File: #169_Majority_Element/Solution.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def majorityElement(self, nums):
    	major = 0
    	count = 0
    	for n in nums:
    		if count == 0:
    			major = n
    		if n == major:
    			count += 1
    		else:
    			count -= 1
    	return major

sol = Solution()
print sol.majorityElement([1, 2, 3, 1, 1])